File: 16 November 2022/Python/Sum of Beauty of All Substrings.py
#User function Template for python3

class Solution:
    def beautySum(self, s):
        result=0
        for i in range(len(s)):
            S={}
            for j in range(i,len(s)):
                if s[j] not in S:
                    S[s[j]]=1
                else:
                    S[s[j]]+=1
                mostFrequency,leastFrequency=max(S.values()),min(S.values())
                result+=(mostFrequency-leastFrequency)
        return result

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        s = input()
        ob = Solution()
        print(ob.beautySum(s))
# } Driver Code Ends

File: November 2022/16 November 2022/Java/Sum of Beauty of All Substrings.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t;
        t = Integer.parseInt(br.readLine());
        while(t-- > 0){
            //br.readLine();
            String s;
            s = br.readLine();
            
            Solution obj = new Solution();
            int res = obj.beautySum(s);
            
            System.out.println(res);
            
        }
    }
}

// } Driver Code Ends


class Solution {
    public static int beautySum(String s) {
        int ans = 0;
        int len = s.length();
        
        for(int start = 0; start < len; start++){
            int[] arr = new int[26];
            for(int end = start; end < len; end++){
                arr[s.charAt(end) - 'a']++;
                
                int max = Integer.MIN_VALUE;
                int min = Integer.MAX_VALUE;
                
                for(int k = 0; k < 26; k++){
                    max = Math.max(max, arr[k]);
                    if(arr[k] != 0){
                        min = Math.min(min, arr[k]);
                    }
                }
                ans += max - min;
            }
        }
        
        return ans;
    }
}
        
